### PR TITLE
Week three step five tutorial

### PR DESCRIPTION
**Pull request by panerorenn9541**
Finish up to step five of the week three tutorial.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -25,7 +33,54 @@
...
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJson(comments));
+  }
+
+  private String convertToJson(ArrayList<Comment> cmts) {
+    Gson gson = new Gson();
+    String json = gson.toJson(cmts);
+    return json;
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: To improve readability, it's better to organize your code to align public methods together, and private methods together.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -25,7 +33,54 @@
...
+  }
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: To improve readability, it's better to organize your code to align public methods together, and private methods together.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -25,7 +33,54 @@
...
+    datastore.put(messageEntity);
+
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Maybe rename this to getRequestParameterOrDefault to be clearer?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.google.sps.data;
+
+public final class Comment {
+  private final String message;
+
+  public Comment(String message) {
+    this.message = message;
+  }
+}
```
Comment (huzcn) [inline portfolio/src/main/java/com/google/sps/data/Comment.java]: Why do you put this class in a separate file? If it's only in DataServlet.java, it's better to be created there. Besides, if this comment is only a string literal, there is no need to define this class.

**Review by huzcn — APPROVED** [review]
Leave a few comments primarily in java code.

**Review by huzcn — CHANGES_REQUESTED** [review]
changes needed before merging

**Comment by panerorenn9541** [comment]
Fix please take a look

**Comment by panerorenn9541** [comment]
Fix and commit changes. Also, when I press "View changes" nothing comes up, but if you click on "Commits" at the top and go to "Fix suggested changes" from July 7, the changes are there.

**Review by huzcn — APPROVED** [review]
Nice!

**Pull request merged**